### PR TITLE
Moved init.sh to nnf-deploy subcommand

### PR DESCRIPTION
**Pull request by bdevcich-hpe**
- Removed init.sh; replaced by ./nnf-deploy init
- Nodes to receive labels/taint are defaulted to workers and rabbits, but can now be overridden via `config/systems.yaml`
- Applied formatting to `config/systems.yaml`
- Updated Readme.md

Signed-off-by: Blake Devcich <blake.devcich@hpe.com>

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -28,12 +28,13 @@ import (
...
-	Name     string                    `yaml:"name"`
-	Aliases  []string                  `yaml:"aliases,flow,omitempty"`
-	Overlays []string                  `yaml:"overlays,omitempty,flow"`
-	Master   string                    `yaml:"master"`
-	Workers  []string                  `yaml:"workers,flow,omitempty"`
-	Rabbits  map[string]map[int]string `yaml:"rabbits,flow"`
+	Name      string                    `yaml:"name"`
+	Aliases   []string                  `yaml:"aliases,flow,omitempty"`
+	Overlays  []string                  `yaml:"overlays,omitempty,flow"`
+	Master    string                    `yaml:"master"`
```
Comment (ajfloeder) [inline config/config.go:34]: I think we can have more than 1 master

File: Readme.md [hunk 2]
```diff
@@ -54,6 +54,9 @@ Flags:
       --dry-run    Show what would be run.
 
 Commands:
+  init
+    Initialize cluster nodes with labels/taints and install cert manager.
```
Comment (NateThornton) [inline Readme.md:58]: I'd make this less technical "Initialize cluster" is enough.

File: config/config.go (hunk 1 above)
Comment (NateThornton) [reply config/config.go:34]: Master isn't even used, so we could get rid of it (in another PR).

File: main.go [hunk 3]
```diff
@@ -429,6 +430,100 @@ func (cmd *InstallCmd) Run(ctx *Context) error {
 	})
 }
 
+type InitCmd struct {
+	Unused string
```
Comment (NateThornton) [inline main.go:434]: Why is this needed, does an empty struct not work?

File: Readme.md [hunk 4]
```diff
@@ -67,9 +70,47 @@ Commands:
...
+
+- `cray.nnf.manager=true`
+- `cray.wlm.manager=true`
+
+Additionally, the NNF nodes (defaults to the rabbit nodes) will obtain the `"cray.nnf.node=true"`
+label and the `"cray.nnf.node=true:NoSchedule"` taint.
 
+These labels/taint will be applied using the `--overwrite=true` option to `kubectl`.
+
+The nodes which receive these labels/taint can be overridden in `config/systems.yaml` with the
```
Comment (roehrich-hpe) [inline Readme.md:92]: Your use of the word "override" is throwing me off.  The systems.yaml is the _the_config_.  Without it, you don't know about the nodes or how to treat each one.

File: config/systems.yaml [hunk 5]
```diff
@@ -1,41 +1,50 @@
...
+  - name: kind
+    aliases: [kind-kind]
+    overlays: [kind, overlays/kind]
+    master: kind-control-plane
+    workers: [kind-worker]
+    rabbits:
+      kind-worker2: { 0: compute-01, 1: compute-02, 6: compute-03 }
+      kind-worker3: { 4: compute-04 }
+    overrides:
+      managers: [kind-control-plane]
```
Comment (roehrich-hpe) [inline config/systems.yaml:11]: What does "manager" mean to you?

Comment (NateThornton) [reply config/systems.yaml:11]: cray.(nnf|wlm).manager=true

Comment (roehrich-hpe) [reply config/systems.yaml:11]: The nodes in the 'rabbits' list should be labeled with cray.nnf.node=true and tainted.  The nodes in the 'worker's list get the other label.

That should work for everything.  You shouldn't need the "overrides" section.

File: Readme.md [hunk 6]
```diff
@@ -67,9 +70,47 @@ Commands:
...
+    aliases: [kind-kind]
+    overlays: [kind, overlays/kind]
+    master: kind-control-plane
+    workers: [kind-worker]
+    rabbits:
+      kind-worker2: { 0: compute-01, 1: compute-02, 6: compute-03 }
+      kind-worker3: { 4: compute-04 }
+    overrides:
+      managers: [kind-control-plane]
+      nnfNodes: [kind-worker3]
```
Comment (NateThornton) [inline Readme.md:106]: I don't think we have a use case where nnf nodes need any sort of override. It should always be the keys to the System.Rabbits map. (Maybe I'm missing it?)

File: config/systems.yaml (hunk 5 above)
Comment (NateThornton) [reply config/systems.yaml:11]: Agreed, and that is the default behavior, but current init.sh has some exceptions to that rule
dp0: worker nodes are compute nodes
craystack: worker nodes are master (control-plane) nodes

Comment (roehrich-hpe) [reply config/systems.yaml:11]: The current systems.yaml for craystack shows this:
```
- name: craystack-default
  aliases: [craystack]
  overlays: [craystack]
  master: k3s-master
  workers: [k3s-master]
```
listing the k3s-master as both master and worker.  So there shouldn't be a need for an "override" section.

Comment (NateThornton) [reply config/systems.yaml:11]: Now I'm catching up with you. Agreed - no need for an override section.

dp0 is _fine_ with just cn-02 as a worker, we could add cn-01 to systems.yaml. 
cn-03 is down and cn-04 is the master so we can ignore those two

Comment (bdevcich-hpe) [reply config/systems.yaml:11]: So for kind, if we get rid of overrides and then set `workers: [kind-control-plane]`, then `kind-worker` is unaccounted for. Do we make that a rabbit?

File: config/config.go (hunk 1 above)
Comment (bdevcich-hpe) [reply config/config.go:34]: I can remove it.

File: config/systems.yaml (hunk 5 above)
Comment (NateThornton) [reply config/systems.yaml:10]: My experience with Kubernetes v1.25.2 was cert-manager needs an untainted pod as the worker; so we should leave the workers value, `workers: [kind-worker]`, alone.

**Review by NateThornton — APPROVED** [review]
